src/graphql/directive/mongoose_graphql_directive_use.erl

```diff
@@ -128,8 +138,32 @@ filter_unloaded_services(Services) ->
     lists:filter(fun(S) -> not mongoose_service:is_loaded(binary_to_existing_atom(S)) end,
                  Services).
 
--spec resolve_not_loaded_fun([binary()], [binary()]) -> resolver().
-resolve_not_loaded_fun(Modules, Services) ->
-    Msg = <<"Some of required modules or services are not loaded">>,
-    Extra = #{not_loaded_modules => Modules, not_loaded_services => Services},
+-spec filter_unloaded_db([binary()]) -> [binary()].
+filter_unloaded_db(DBs) ->
+    lists:filter(fun(DB) -> is_database_unloaded(DB) end, DBs).
+
+-spec is_database_unloaded(binary()) -> boolean().
+is_database_unloaded(DB) ->
+    mongoose_config:lookup_opt([internal_databases,
+                                binary_to_existing_atom(DB)]) == {error, not_found}.
+
+-spec resolve_not_loaded_fun([{dependency_type(), [binary()]}]) -> resolver().
+resolve_not_loaded_fun(NotLoaded) ->
+    Msg = not_loaded_message(NotLoaded),
+    Extra = maps:from_list([{error_key(Item), L} || {Item, L} <- NotLoaded]),
     fun(_, _, _, _) -> mongoose_graphql_helper:make_error(deps_not_loaded, Msg, Extra) end.
+
+-spec not_loaded_message([{dependency_type(), [binary()]}]) -> binary().
+not_loaded_message(NotLoaded) ->
+    MsgPrefix = <<"Some of the required ">>,
+    MsgList = string:join([item_to_string(Item) || {Item, _} <- NotLoaded], " and "),
+    MsgBinaryList = list_to_binary(MsgList),
+    <<MsgPrefix/binary, MsgBinaryList/binary, " are not loaded">>.
+
+-spec item_to_string(dependency_type()) -> [string()].
+item_to_string(Item) ->
```

Comment:
Item is too generic (it could be confusing that it is dependency type not dependency name).
Just reuse type names when naming functions:

```suggestion
dependency_type_to_string(Type) ->
```

And below too, use Type  instead of Item.